interfaces/system_key.go

```diff
@@ -172,8 +175,19 @@ func UnmarshalJSONSystemKey(r io.Reader) (interface{}, error) {
 	return sk, nil
 }
 
+// SystemKeyExtraData holds information about the current state of the system
+// key so that some values do not need to be re-checked and can thus be
+// guaranteed to be consistent across multiple uses of system key functions.
+type SystemKeyExtraData struct {
+	// PromptingFlagEnabled indicates whether the AppArmorPrompting feature
+	// flag is currently set to true. This value is ANDed with the prompting
+	// supported value, so setting this value to supported&&enabled results
+	// in an identical system key as would be generated using enabled alone.
+	PromptingFlagEnabled bool
```

Comment:
if enabled is true but support is false then enabled is true but supported&enabled is false, so the last sentence isn't true

Reply:
The system key stores `supported` and `supported&&enabled`, so the resulting system key is identical regardless of whether `enabled` or `supported&&enabled` is used.

The goal is to avoid needing to call `features.AppArmorPrompting.IsSupported()` unless absolutely necessary, and to avoid regenerating profiles unless absolutely necessary. So by storing `supported` we can omit it when `generateSystemKey` and rely on the parser mtime, and if AA features change (which is the only way prompting support can change), then we'll regenerate profiles anyway. And by storing `supported&&enabled`, if we change the flag but support is unsupported and that support is unchanged (which we know because the parser mtime and kernel features are unchanged, without needing to probe parser features again), we don't need to regenerate profiles because we still know prompting will not be used. And if prompting is supported, since `supported&&enabled` is stored, that value will be different in the system key so profiles will be regenerated.

Reply:
Perhaps this is the confusion:

> This value is ANDed with the prompting supported value

should become

> When generating the system key, this value will be ANDed with the prompting supported value